pkg/api/v1/pod/util.go

```diff
@@ -50,38 +50,65 @@ func FindPort(pod *v1.Pod, svcPort *v1.ServicePort) (int, error) {
 	return 0, fmt.Errorf("no suitable port for manifest: %s", pod.UID)
 }
 
+// ContainerType signifies container type
+type ContainerType int
+
+// DefaultContainers defines default behavior: Iterate containers based on feature gates
+const DefaultContainers ContainerType = 0
```

Comment:
Suggest getting rid of this and using AllContainers

Reply:
done. since the changes are large and irrelevant, i moved them to a separate PR https://github.com/kubernetes/kubernetes/pull/89308.